python/samples/04.ai.g.autogen.product-spec-critique/src/state.py

```diff
@@ -0,0 +1,41 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from typing import Optional, List, Dict
+
+from botbuilder.core import Storage, TurnContext
+from teams.state import ConversationState, TempState, TurnState, UserState
+from datetime import datetime
+
+class AppConversationState(ConversationState):
+    message_history: List[Dict] | None = None
```

Comment:
pipe or syntax won't work in older versions of python, please switch to use `Union` instead for backwards compat.